subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c

```diff
@@ -20,9 +20,21 @@
 #include <nrfx/hal/nrf_rtc.h>
 #include <nrfx/hal/nrf_ccm.h>
 
+/* Temporary hacks to make this file compile successfully for boards simulated
+ * with BabbleSim, without updating now nrfx HAL derived files in the BabbleSim
+ * repository.
+ */
+#if defined(CONFIG_SOC_SERIES_BSIM_NRFXX)
+	#define nrf_radio(op, ...)  nrf_radio_##op(__VA_ARGS__)
+	#define nrf_timer_cc_set    nrf_timer_cc_write
+#else
+	#define nrf_radio(op, ...)  nrf_radio_##op(NRF_RADIO, __VA_ARGS__)
+#endif
+
 #if defined(CONFIG_SOC_SERIES_NRF51X)
 #define RADIO_PDU_LEN_MAX (BIT(5) - 1)
-#elif defined(CONFIG_SOC_COMPATIBLE_NRF52X)
+#elif defined(CONFIG_SOC_COMPATIBLE_NRF52X) || \
```

Comment:
make it `#else`?

Reply:
Related to your above comment, @joerchan. We should probably have an ifdef here coupled with the "capability" for DLE.

But for now, I would prefer to leave as is, cause this is more **educative**, about which platforms really can support DLE.

Reply:
Yes, I agreed. We should send a refactoring PR with DLE feature support. But not as part of this PR.

Reply:
@ioannisg I would agree with you, but this is not about supporting DLE. This is about supporting an older chip that was made when the length field of the LL PDU was only 5 bits. The length field is gonna be 8 bits whether or not you support DLE or not. Which is why I think my suggestion make sense, and is easier to maintain. (Same for CCM Extended length).

Reply:
> We should send a refactoring PR with DLE feature support. But not as part of this PR.

Refactoring the existing codebase to be able to support a new feature should be a part the feature-PR.

Reply:
Right, I see what you mean - you want the length of the field to be tight to the SoC and not the larger PDU length extension. Then, I 'll send a PR upstream and remove the change from here.

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/20304
If the above goes through, then this will be automatically resolved. Otherwise, we keep current solution.